docs/preview/features/correlation.md

```diff
@@ -0,0 +1,206 @@
+            // otherwise a GUID will be generated (default: true).
+            options.Transaction.GenerateWhenNotSpecified = true;
+
+            // Whether to include the transaction ID in the response (default: true).
+            options.Transaction.IncludeInResponse = true;
+
+            // The header to look for in the HTTP request, and will be set in the HTTP response (default: X-Transaction-ID).
+            options.Transaction.HeaderName = "X-Transaction-ID";
+
+            // The function that will generate the transaction ID, when the `.GenerateWhenNotSpecified` is set to `false` and the request doesn't contain the header.
+            // (default: new `Guid`).
+            options.Transaction.GenerateId = () => $"Transaction-{Guid.NewGuid()}";
+
+            // Configuration on the operation ID (`RequestId`) response header.
+            // ----------------------------------------------------------------
+
+            // Whether to include the operation ID in the response (default: true).
+            options.Operation.IncludeInResponse = true;
+
+            // The header that will contain the operation ID in the HTTP response (default: RequestId).
+            options.Operation.HeaderName = "RequestId";
+
+            // The function that will generate the operation ID header value.
+            // (default: new `Guid`).
+            options.Operation.GenerateId = () => $"Operation-{Guid.NewGuid()}";
+
+            // Configuration on operation parent ID request header (`Request-Id`).
+            // ------------------------------------------------------------------
+
+            // Whether to extract the operation parent ID from the incoming request following W3C Trace-Context standard (default: true).
+            // More information on operation ID and operation parent ID, see [this documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation).
+            options.UpstreamService.ExtractFromRequest = false;
+
+            // The header that will contain the operation parent ID in the HTTP request (default: Request-Id).
+            options.UpstreamService.OperationParentIdHeaderName = "x-request-id";
+        });
+    }
+}
+```
+
+## Logging
+
+As an additional feature, we provide an extension to use the HTTP correlation directly in a [Serilog](https://serilog.net/) configuration:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Serilog;
+
+public class Startup
+{
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseHttpCorrelation();
+        
+        Log.Logger = new LoggerConfiguration()
+            .Enrich.WithHttpCorrelationInfo()
+            .WriteTo.Console()
+            .CreateLogger();
+    }
+}
+```
+
+## Using secret store within Azure Functions
```

Comment:
Does it make sense to have this section here in a document regarding correlation ?
I think the title should be different ?  (Using correlation within Azure Functions or something like that ? )